cmd/jaeger/internal/extension/jaegerstorage/extension.go

```diff
@@ -116,7 +116,7 @@ func (s *storageExt) Start(ctx context.Context, host component.Host) error {
 		cfg:         s.config.Badger,
 		builder:     badger.NewFactoryWithConfig,
 	}
-	grpcStarter := &starter[grpcCfg.ConfigV2, *grpc.Factory]{
+	grpcStarter := &starter[grpcCfg.ConfigurationV2, *grpc.Factory]{
```

Comment:
a) I think `config.ConfigV2` was fine, not very different from stdlib `context.Context`

b) a better fix is to remove grpc/config package and just move the config into grpc/ directly. I don't see any usage that would break because of that:
```
$ grx plugin/storage/grpc/config .
Binary file ./cmd/jaeger/jaeger matches
./cmd/jaeger/internal/extension/jaegerstorage/config.go:14:	grpcCfg "github.com/jaegertracing/jaeger/plugin/storage/grpc/config"
./cmd/jaeger/internal/extension/jaegerstorage/extension.go:23:	grpcCfg "github.com/jaegertracing/jaeger/plugin/storage/grpc/config"
./plugin/storage/grpc/options.go:26:	"github.com/jaegertracing/jaeger/plugin/storage/grpc/config"
./plugin/storage/grpc/factory.go:30:	"github.com/jaegertracing/jaeger/plugin/storage/grpc/config"
./plugin/storage/grpc/options_test.go:26:	grpccfg "github.com/jaegertracing/jaeger/plugin/storage/grpc/config"
./plugin/storage/grpc/factory_test.go:37:	grpcConfig "github.com/jaegertracing/jaeger/plugin/storage/grpc/config"
```